extension/wallet/src/protocol/repay_loan.rs

```diff
@@ -137,13 +152,8 @@ pub async fn repay_loan(
                 // here. Maybe we should move it to a common
                 // place, so it can be used for different
                 // protocols
-                let script_witness = sign_with_key(
-                    SECP256K1,
-                    &mut cache,
-                    index,
-                    &wallet.secret_key,
-                    output.value,
-                );
+                let script_witness =
+                    sign_with_key(SECP256K1, &mut cache, index, &secret_key, output.value);
```

Comment:
Can we add some form of signing API to `Wallet` that reduces the amount of code we have to write here?